386 header cart api fetch

kumarshoolam opened the pull request:
Fix #386 

Test URLs:
- Before: https://main--danaher-abcam--aemsites.aem.live/en-us/webinars/webinar-two
- After: https://386-headercart-api-fetch--danaher-abcam--aemsites.aem.live/en-us/webinars/webinar-two

pardeepgera23 commented on eds/blocks/header/header.js:
```diff
@@ -419,4 +419,45 @@ export default async function decorate(block) {
+    })
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error(`HTTP error! Status: ${response.status}`);
```
log error in console

kumarshoolam replied:
Hello @pardeepgera23 
Addressed PR Review Comments & Made Changes Accordingly-386.
CC: @Meenu-GSPANN, @Rajesh-GSPANN

pardeepgera23 replied:
Use // eslint-disable-next-line no-console

pardeepgera23 commented on eds/blocks/header/header.js:
```diff
@@ -419,4 +419,45 @@ export default async function decorate(block) {
+  // Cart icon
+  const hostName = (!window.location.host.includes('localhost') && !window.location.host.includes('.hlx'))
+    ? window.location.host
+    : 'pp.abcam.com';
```
no need to point to pp.abcam.com. Use window.location.host only.

kumarshoolam replied:
Hello @pardeepgera23 
Addressed PR Review Comments & Made Changes Accordingly-386.
CC: @Meenu-GSPANN, @Rajesh-GSPANN

pardeepgera23 reviewed (changes requested):
added comments

kumarshoolam commented on eds/blocks/header/header.js:
```diff
@@ -419,4 +419,45 @@ export default async function decorate(block) {
+    fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
```
Hello @pardeepgera23 
Addressed PR Review Comments & Made Changes Accordingly-386.
CC: @Meenu-GSPANN, @Rajesh-GSPANN

pardeepgera23 commented on eds/blocks/header/header.js:
```diff
@@ -419,4 +428,51 @@ export default async function decorate(block) {
+      window.location.href = `https://${hostName}/en-us/shopping-basket/${shoppingBaskedId}?country=${selectedCountry}`;
+    } else {
+      // eslint-disable-next-line no-bitwise
+      alert(`You have ${productCount} products in your cart!`);
```
Either use modal popup instead of alert or ignore the eslint for alert using // eslint-disable-next-line no-alert

rgravitvl reviewed (approved):
Please address the redirect empty cart page, x-correlation-id and rest looks good to me

kumarshoolam commented:
Hello @rgravitvl 

As we discussed, We have Addressed PR Comments regarding empty cart page redirection, x-correlation-id and worked accordingly.

CC: @Meenu-GSPANN , @pardeepgera23 , @Rajesh-GSPANN
